packages/flutter_tools/tool/tool_coverage.dart

```diff
@@ -3,165 +3,164 @@
-  // This will set up several fields used by the test methods on the context. Normally
-  // this would be handled automatically by the test runner, but since we're executing
-  // the files directly with dart we need to handle it manually.
-  String _createTest(File testFile) {
-    final File fakeTest = File(path.join(tempDirectory.path, testFile.path))
-      ..createSync(recursive: true)
-      ..writeAsStringSync('''
-import "package:test/test.dart";
-import "${path.absolute(testFile.path)}" as entrypoint;
+  /// Spawns an isolate and passes it [message].
+  ///
+  /// This isolate connects an [IsolateChannel] to [message] and sends the
+  /// serialized tests over that channel.
+  Future<Isolate> _spawnIsolate(String path, SendPort message) async {
+    return _spawnPrecompiledIsolate(path, message, precompiledPath);
+  }
 
-void main() {
-  group('', entrypoint.main);
-}
-''');
-    return fakeTest.path;
+  Future<Isolate> _spawnPrecompiledIsolate(String testPath, SendPort message, String precompiledPath) async {
+    testPath = p.absolute(p.join(precompiledPath, testPath) + '.vm_test.dart');
+    testPath = testPath.substring(0, testPath.length - '.dart'.length) + '.vm.app.dill';
+    return await Isolate.spawnUri(p.toUri(testPath), <String>[], message,
+      packageConfig: p.toUri('.packages'),
+      checked: true,
+    );
   }
 
-  Future<void> _runTest(File testFile) async {
-    final PoolResource resource = await pool.request();
-    final String testPath = _createTest(testFile);
-    final int port = await _findPort();
-    final Uri coverageUri = Uri.parse('http://127.0.0.1:$port');
-    final Completer<void> completer = Completer<void>();
-    final String packagesPath = argResults['packages'];
-    final Process testProcess = await Process.start(
-      Platform.resolvedExecutable,
-      <String>[
-        '--packages=$packagesPath',
-        '--pause-isolates-on-exit',
-        '--enable-asserts',
-        '--enable-vm-service=${coverageUri.port}',
-        testPath,
-      ],
-      runInShell: true,
-      environment: <String, String>{
-        'FLUTTER_ROOT': flutterRoot,
-      }).timeout(const Duration(seconds: 30));
-    testProcess.stdout
-      .transform(utf8.decoder)
-      .transform(const LineSplitter())
-      .listen((String line) {
-        print(line);
-        if (line.contains('All tests passed') || line.contains('Some tests failed')) {
-          completer.complete(null);
-        }
-      });
+  @override
+  Future<void> close() async {
     try {
-      await completer.future;
-      await coverageCollector.collectCoverage(testProcess, coverageUri).timeout(const Duration(seconds: 30));
-      testProcess?.kill();
+      await Future.wait(_pending.values).timeout(const Duration(seconds: 10));
     } on TimeoutException {
-      print('Failed to collect coverage for ${testFile.path} after 30 seconds');
-    } finally {
-      resource.release();
+      // Do nothing.
```

Comment:
Would it help debug the underlying issue to add some logging here?

Reply:
Done, it seems this happens when underlying test times out